tests/components/backblaze_b2/conftest.py

```diff
@@ -0,0 +1,368 @@
+            # Add attributes that match real Bucket class for diagnostics compatibility
+            self.name = self.bucket_name
+            self.id_ = self.bucket_id
+            self.type_ = self.bucket_type
+            self.cors_rules = []
+            self.lifecycle_rules = []
+            self.revision = 1
+
+        BucketSimulator.__init__ = patched_init
+
+        # Mock start_large_file and cancel_large_file for BucketSimulator
+        def mock_start_large_file(
+            file_name, content_type, file_info, account_auth_token
+        ):
+            del (
+                content_type,
+                file_info,
+                account_auth_token,
+            )  # Required by interface but not used
+            mock_large_file = Mock()
+            mock_large_file.file_name = file_name
+            mock_large_file.file_id = "mock_file_id"
+            return mock_large_file
+
+        def mock_cancel_large_file(file_id, account_auth_token):
+            del file_id, account_auth_token  # Required by interface but not used
+
+        BucketSimulator.start_large_file = mock_start_large_file
+        BucketSimulator.cancel_large_file = mock_cancel_large_file
+
+        # Mock upload_local_file for the Bucket object - needed for direct file uploads
+        def mock_upload_local_file(
+            local_file,
+            file_name,
+            content_type=None,
+            file_info=None,
+            progress_listener=None,
+        ):
+            del progress_listener  # Required by interface but not used
+            # Read the file content
+            with open(local_file, "rb") as f:
+                content = f.read()
+
+            # Use the existing upload_file mock logic
+            stream = io.BytesIO(content)
+            stream.seek(0)
+            return sim.upload_file(
+                upload_url["uploadUrl"],
+                upload_url["authorizationToken"],
+                file_name,
+                content_length=len(content),
+                content_type=content_type or "application/octet-stream",
+                content_sha1=None,
+                file_info=file_info or {},
+                data_stream=stream,
+            )
+
+        # We need to patch the actual Bucket class, not BucketSimulator
+        import b2sdk.v2.bucket  # noqa: PLC0415
+
+        original_upload_local_file = getattr(
+            b2sdk.v2.bucket.Bucket, "upload_local_file", None
+        )
+        b2sdk.v2.bucket.Bucket.upload_local_file = mock_upload_local_file
+
+        try:
+            yield BackblazeFixture(
+                application_key_id, application_key, bucket, sim, auth
+            )
+        finally:
+            # Restore original method if it existed
+            if original_upload_local_file is not None:
+                b2sdk.v2.bucket.Bucket.upload_local_file = original_upload_local_file
+            else:
+                delattr(b2sdk.v2.bucket.Bucket, "upload_local_file")
```

Comment:
What's going on here, why are we monkey patching the library instead of using `unittest.patch`?